legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/serialization/filesystem/usercodestorage/CodeStorage.java

```diff
@@ -75,6 +75,11 @@
      */
     boolean isVersioned(String path);
 
+    default boolean isRepositoryImmutable(String name)
```

Comment:
I tend to think this should be on MutableCodeStorage, since it applies to questions of mutability.